test/domain_bridge/test_domain_bridge.cpp

```diff
@@ -0,0 +1,179 @@
+
+#include "domain_bridge/domain_bridge.hpp"
+
+class TestDomainBridge : public ::testing::Test
+{
+protected:
+  static void SetUpTestCase()
+  {
+    rclcpp::init(0, nullptr);
+  }
+
+  static void TearDownTestCase()
+  {
+    rclcpp::shutdown();
+  }
+};
+
+TEST_F(TestDomainBridge, construction_destruction)
+{
+  // Default
+  {
+    domain_bridge::DomainBridge bridge;
+  }
+  // With options
+  {
+    domain_bridge::DomainBridge bridge(domain_bridge::DomainBridgeOptions());
+  }
+}
+
+TEST_F(TestDomainBridge, get_options)
+{
+  domain_bridge::DomainBridgeOptions input_options;
+  input_options.name("domain_bridge_test_name");
+  domain_bridge::DomainBridge bridge(input_options);
+  domain_bridge::DomainBridgeOptions output_options = bridge.get_domain_bridge_options();
+  EXPECT_EQ(input_options.name(), output_options.name());
+}
+
+TEST_F(TestDomainBridge, bridge_topic_valid)
+{
+  // Nominal test
+  {
+    domain_bridge::DomainBridge bridge;
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 2);
+  }
+  // Same domain ID
+  {
+    domain_bridge::DomainBridge bridge;
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 1);
+  }
+  // Topic with namespace
+  {
+    domain_bridge::DomainBridge bridge;
+    bridge.bridge_topic("foo/bar/baz", "test_msgs/msg/BasicTypes", 1, 2);
+  }
+  // Multiple topics
+  {
+    domain_bridge::DomainBridge bridge;
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 2);
+    bridge.bridge_topic("bar", "test_msgs/msg/BasicTypes", 1, 2);
+    bridge.bridge_topic("baz", "test_msgs/msg/BasicTypes", 1, 2);
+  }
+  // Same topic, different domains
+  {
+    domain_bridge::DomainBridge bridge;
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 2);
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 3);
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 2, 1);
+  }
+  // Same topic, different types
+  // This use-case isn't clearly supported by ROS 2, see https://github.com/ros2/ros2/issues/1095
+  // {
+  //   domain_bridge::DomainBridge bridge;
+  //   bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 2);
+  //   bridge.bridge_topic("foo", "test_msgs/msg/Nested", 1, 2);
+  //   bridge.bridge_topic("foo", "test_msgs/msg/Constants", 1, 2);
+  // }
+}
+
+TEST_F(TestDomainBridge, bridge_topic_invalid)
+{
+  // Invalid topic name
+  {
+    domain_bridge::DomainBridge bridge;
+    EXPECT_THROW(
+      bridge.bridge_topic("Not a v@lid topic name!", "test_msgs/msg/BasicTypes", 1, 2),
+      rclcpp::exceptions::InvalidTopicNameError);
+  }
+  // Invalid type name
+  {
+    domain_bridge::DomainBridge bridge;
+    EXPECT_THROW(bridge.bridge_topic("foo", "not a valid message type", 1, 2), std::runtime_error);
+  }
+  // Same bridge twice
+  {
+    testing::internal::CaptureStderr();
+    domain_bridge::DomainBridge bridge;
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 2);
+    bridge.bridge_topic("foo", "test_msgs/msg/BasicTypes", 1, 2);
```

Comment:
shouldn't this throw an exception?

Reply:
Bridging the same topic/type for the same domains is essentially a noop. I originally was throwing an exception, though after your suggestion in https://github.com/ros2/domain_bridge/pull/1#discussion_r583752870, I switched to a warning, which seems reasonable to me since I don't think it is going to cause any severe errors.

Reply:
Sounds fine to me.
I think my original comment was because the first implementation didn't support bridging the same topic to two different domains, but I don't mind much.